gen/cycle.ml

```diff
@@ -463,13 +463,18 @@ module CoSt = struct
 
   let get_cell st = st.co_cell
 
-  let set_cell st e = match e.bank with
-    | Ord ->
+  let set_cell st n =
+    let e = n.evt in match e.bank with
+    | Ord -> begin
        let old = st.co_cell.(0) in
        let cell = E.overwrite_value old e.atom e.v in
        let co_cell = Array.copy st.co_cell in
-       co_cell.(0) <- cell ;
-       {e with cell=co_cell;},{ st with co_cell; }
+       let cell2 = match n.prev.edge.E.edge with
+        | E.Rmw rmw -> E.compute_rmw rmw old e.v
+        | _ -> cell in
```

Comment:
Shouldn't this be:
```
| _ -> e.v
```
In which case we can also get rid of line 470?

Reply:
Hi @relokin, I usually merge with a merge commit. If commit history is complex I squash. I mean I use github "squash and merge" button. Here I do not know yet.

Reply:
Hi @hugookeeffe and @relokin, I'd like to merge the PR in once the last comment by @relokin above is handled. 

Notice that, once simplified, the `set_cell` function can be adapted easily to mixed-size by using the `E.extract_value` function. For clarity, commit ba305dfc1eb3ffd57087c04e57b6def2c65eee6a performs the adaptation (also addressing @relokin's comment).

Reply:
Hi @maranget, I have made the changes for the last comment in my recent commit. Thank you for the help!

Reply:
Thanks, I'll merge tomorrow.